src/semantics/expression_builder.ts

```diff
@@ -921,6 +923,14 @@ function buildArrayLiteralExpression(
 
     // return new NewLiteralArrayValue(array_type!, init_values);
     const elem_type = (array_type as ArrayType).element;
+    // Workaround: solve the case that the return value of a generic function is a generic array
```

Comment:
Why this only work for `return value of a generic function`? Seems it should also work for assignment statement?

Reply:
done.
In **buildArrayLiteralExpression**, I made common logic to process generic arrays.It does not only work for return value of a generic function.